### PR TITLE
Disable Advance Day Button While Advance Day is in Progress

### PR DESCRIPTION
**Pull request by IllianiCBT**
Added logic to disable the Advance Day button to prevent multiple activations while the advance day event is still being resolved. Re-enabled the button after the process completes, ensuring proper functionality and avoiding unintended behavior.

### CODE REVIEW

File: MekHQ/src/mekhq/gui/CampaignGUI.java [hunk 1]
```diff
@@ -1623,21 +1635,8 @@ public void refitUnit(Refit r, boolean selectModelName) {
...
-                    .append(tech.getPrimaryRoleDesc())
-                    .append(" (")
-                    .append(getCampaign().getTargetFor(r, tech).getValueAsString())
-                    .append("+), ")
-                    .append(tech.getMinutesLeft())
-                    .append('/')
-                    .append(tech.getDailyAvailableTechTime())
-                    .append(" minutes</html>");
-                name = nameBuilder.toString();
+                String nameBuilder = "<html>" + tech.getFullName() + ", <b>" + SkillType.getColoredExperienceLevelName(tech.getSkillLevel(getCampaign(), false)) + "</b> " + tech.getPrimaryRoleDesc() + " (" + getCampaign().getTargetFor(r, tech).getValueAsString() + "+), " + tech.getMinutesLeft() + '/' + tech.getDailyAvailableTechTime() + " minutes</html>";
```
Comment (Sleet01) [inline MekHQ/src/mekhq/gui/CampaignGUI.java:1638]: This seems less clean, plus if we're not using a StringBuilder you can just concatenate everything into `name` and skip the second line.

Was this an IDE-suggested change?  Apparently most will turn this form into the StringBuilder form under the hood, so I'm not sure this change gets us anything.

**Review by Sleet01 — CHANGES_REQUESTED** [review]
I'm not sure about the nameBuilder change; do you mind giving the rationale for that?

Otherwise the fix looks good.

**Comment by IllianiCBT** [comment]
That was IDEA 'helping'

**Comment by IllianiCBT** [comment]
> I'm not sure about the nameBuilder change; do you mind giving the rationale for that?

Should be good now. Glad you caught that. IDEA launched a new version and their file clean-up in previous versions was less aggressive. Disabled it moving forward.

**Review by Sleet01 — APPROVED** [review]
LGTM!

**Pull request merged**